Docs/Book/GettingStartedInC++.md

```diff
@@ -0,0 +1,1420 @@
+
+It is also possible to produce a 2D picture that attempts to mimic as
+closely as possible a 3D conformation.  Again, an equivalent of the
+Python function
+`rdkit.Chem.AllChem.GenerateDepictionMatching3DStructure` was
+incorporated in December 2016.
+
+### Working with 3D Molecules
+
+The RDKit can generate conformations for molecules using two different
+methods.  The original method used distance geometry [[1]](#blaney).
+The algorithm followed is:
+
+1. The molecule's distance bounds matrix is calculated based on the
+   connection table and a set of rules. 
+
+2. The bounds matrix is smoothed using a triangle-bounds smoothing
+   algorithm. 
+
+3. A random distance matrix that satisfies the bounds matrix is
+   generated. 
+
+4. This distance matrix is embedded in 3D dimensions (producing
+   coordinates for each atom). 
+
+5. The resulting coordinates are cleaned up somewhat using a crude
+   force field and the bounds matrix. 
+
+Note that the conformations that result from this procedure tend to be
+fairly ugly. They should be cleaned up using a force field.
+This can be done within the RDKit using its implementation of the
+Universal Force Field UFF[[2]](#rappe).
+
+More recently, there is an implementation of the method of Riniker and
+Landrum [[3]](#riniker2) which uses torsion angle preferences from the
+Cambridge Structural Database (CSD) to correct the conformers after
+distance geometry has been used to generate them.  With this method,
+there should be no need to use a minimisation step to clean up the
+structures.
+
+The full process of embedding and optimizing a molecule is easier than
+all the above verbiage makes it sound
+[(example11.cpp)](./C++Examples/example11.cpp):
+```c++
+#include <GraphMol/DistGeomHelpers/Embedder.h>
+#include <GraphMol/ForceFieldHelpers/UFF/UFF.h>
+.
+.
+RDKit::ROMOL_SPTR mol( RDKit::SmilesToMol( "C1CCC1OC" ) );
+RDKit::ROMOL_SPTR mol1( RDKit::MolOps::addHs( *mol ) );
+// Original distance geometry embedding
+RDKit::DGeomHelpers::EmbedMolecule( *mol1 , 0 , 1234 );
+RDKit::UFF::UFFOptimizeMolecule( *mol1 );
+
+// new Riniker and Landrum CSD-based method
+RDKit::ROMOL_SPTR mol2( RDKit::MolOps::addHs( *mol ) );
+RDKit::DGeomHelpers::EmbedMolecule( *mol2 , 0 , 1234 , true , false ,
+                                    2.0 , true , 1 ,
+                                    static_cast<const std::map<int,RDGeom::Point3D> *> ( 0 ) ,
+                                    1e-3 , false , true , true , true );
+```
+In the second call to `EmbedMolecule`, it is the last two `true`
+parameters that enforce the second embedding method.  Apart from the
+random number seed (1234) the other parameters are set to default
+values.  Setting the random number seed to other than the default -1
+ensures that the same conformations are produced each time the code is
+run.  This is convenient when testing to ensure reproducibility of
+results but disguises the inherently non-deterministic nature of the
+algorithm.
```

Comment:
hmmm, I'm not sure I agree with this. The algorithm is completely deterministic once a seed has been selected.